core/chains/evm/gas/rollups/op_l1_oracle.go

```diff
@@ -0,0 +1,440 @@
+	}
+	getL1GasUsedCalldata, err := getL1GasUsedMethodAbi.Pack(OPStackGasOracle_getL1GasUsed, []byte{0x1})
+	if err != nil {
+		panic(fmt.Errorf("failed to parse GasPriceOracle %s() calldata for chain: %s; %w", OPStackGasOracle_getL1GasUsed, chainType, err))
+	}
+
+	getL1FeeMethodAbi, err := abi.JSON(strings.NewReader(GetL1FeeAbiString))
+	if err != nil {
+		panic(fmt.Errorf("failed to parse GasPriceOracle %s() method ABI for chain: %s; %w", OPStackGasOracle_getL1Fee, chainType, err))
+	}
+	getL1FeeCalldata, err := getL1FeeMethodAbi.Pack(OPStackGasOracle_getL1Fee, []byte{0x1})
+	if err != nil {
+		panic(fmt.Errorf("failed to parse GasPriceOracle %s() calldata for chain: %s; %w", OPStackGasOracle_getL1Fee, chainType, err))
+	}
+
+	return &optimismL1Oracle{
+		client:     ethClient,
+		pollPeriod: PollPeriod,
+		logger:     logger.Sugared(logger.Named(lggr, "L1GasOracle(optimismBedrock)")),
+		chainType:  chainType,
+
+		l1OracleAddress:     l1OracleAddress,
+		gasPriceMethod:      gasPriceMethod,
+		l1GasPriceMethodAbi: l1GasPriceMethodAbi,
+		gasCostMethod:       gasCostMethod,
+		l1GasCostMethodAbi:  l1GasCostMethodAbi,
+
+		chInitialised: make(chan struct{}),
+		chStop:        make(chan struct{}),
+		chDone:        make(chan struct{}),
+
+		isEcotoneMethodAbi: isEcotoneMethodAbi,
+
+		l1BaseFeeCalldata:    l1BaseFeeCalldata,
+		isEcotoneCalldata:    isEcotoneCalldata,
+		getL1GasUsedCalldata: getL1GasUsedCalldata,
+		getL1FeeCalldata:     getL1FeeCalldata,
+
+		isEcotone:        false,
+		isEcotoneCheckTs: 0,
+	}
+}
+
+func (o *optimismL1Oracle) Name() string {
+	return o.logger.Name()
+}
+
+func (o *optimismL1Oracle) Start(ctx context.Context) error {
+	return o.StartOnce(o.Name(), func() error {
+		go o.run()
+		<-o.chInitialised
+		return nil
+	})
+}
+func (o *optimismL1Oracle) Close() error {
+	return o.StopOnce(o.Name(), func() error {
+		close(o.chStop)
+		<-o.chDone
+		return nil
+	})
+}
+
+func (o *optimismL1Oracle) HealthReport() map[string]error {
+	return map[string]error{o.Name(): o.Healthy()}
+}
+
+func (o *optimismL1Oracle) run() {
+	defer close(o.chDone)
+
+	t := o.refresh()
+	close(o.chInitialised)
+
+	for {
+		select {
+		case <-o.chStop:
+			return
+		case <-t.C:
+			t = o.refresh()
+		}
+	}
+}
+func (o *optimismL1Oracle) refresh() (t *time.Timer) {
+	t, err := o.refreshWithError()
+	if err != nil {
+		o.SvcErrBuffer.Append(err)
+	}
+	return
+}
+
+func (o *optimismL1Oracle) refreshWithError() (t *time.Timer, err error) {
+	t = time.NewTimer(utils.WithJitter(o.pollPeriod))
+
+	ctx, cancel := o.chStop.CtxCancel(evmclient.ContextWithDefaultTimeout())
+	defer cancel()
+
+	price, err := o.fetchL1GasPrice(ctx)
+	if err != nil {
+		return t, err
+	}
+
+	o.l1GasPriceMu.Lock()
+	defer o.l1GasPriceMu.Unlock()
+	o.l1GasPrice = priceEntry{price: assets.NewWei(price), timestamp: time.Now()}
+	return
+}
+
+func (o *optimismL1Oracle) fetchL1GasPrice(ctx context.Context) (price *big.Int, err error) {
+	return o.GetDAGasPrice(ctx)
```

Comment:
Think we can just copy the stuff inside of `GetDAGasPrice` here since this is all it does now.

Reply:
As per my reply above, this would effect `op_l1_oracle_test.go`

Reply:
True. Ok in that case we can just call `GetDAGasPrice` directly wherever `fetchL1GasPrice` was being called.